### PR TITLE
pkg/git: remove sensitive data when returning errors that will be output

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

When we have some errors when push objects to Git it can output some sensitive data like the GitHub API key, this PR removes those keys, and then we can output with safe.

Also in the pushing tags, it was output twice or from the std.Err and the other from the error from the tooling, this PR update when running the commands to run as silent and not push things to the output.

this is a sample in what we will see in the logs

```
error: failed to push some refs to 'https://git:[REDACTED]@github.com/kubernetes/kubernetes'
```

/assing @saschagrunert @puerco @kubernetes/release-managers 

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
pkg/git: remove sensitive data when returning errors that will be output
```

### CODE REVIEW

File: pkg/git/git.go [hunk 1]
```diff
@@ -1298,3 +1300,14 @@ func (e NetworkError) CanRetry() bool {
...
 	return false
 }
+
+func safeError(err error) error {
+	if err == nil {
+		return nil
+	}
+	// To parse the GitHub ApiKey
+	re := regexp.MustCompile(`[0-9a-zA-Z]{35,40}`)
+	errUpdated := re.ReplaceAll([]byte(err.Error()), []byte("[REDACTED]"))
```
Comment (saschagrunert) [inline pkg/git/git.go:1310]:
```suggestion
	errUpdated := re.ReplaceAllString(err.Error(), "[REDACTED]")
```

File: pkg/git/git.go [hunk 2]
```diff
@@ -1298,3 +1300,14 @@ func (e NetworkError) CanRetry() bool {
 	// Otherwise permanent
 	return false
 }
+
+func safeError(err error) error {
+	if err == nil {
+		return nil
+	}
+	// To parse the GitHub ApiKey
+	re := regexp.MustCompile(`[0-9a-zA-Z]{35,40}`)
```
Comment (saschagrunert) [inline pkg/git/git.go:1309]: Make this please a global var to avoid re-compiling every time.

Comment (cpanato) [reply pkg/git/git.go:1309]: ok

thanks for your review

**Review by saschagrunert — APPROVED** [review]
/lgtm 

Thanks for the quick fix! 🙌

**Comment by puerco** [comment]
Thanks Carlos! 
/honk

**Pull request merged**